test: update tool_wallet coverage for unexpected writes to wallet

jonatack opened the pull request:
1. Create file permission utilities in the functional test framework based on #27850 and #15687.

2. PR #15687 added test coverage in `test/functional/tool_wallet.py` to reproduce unexpected writes to the wallet file, as described in issue https://github.com/bitcoin/bitcoin/issues/15608:

    - wallet tool info unexpectedly writes to the wallet file if the wallet file permissions are read/write
    - wallet tool info raises if the wallet file permissions are read-only

    Update these tests/docs for the following changes:

    - addition of descriptor wallets
    - CI migration away from Appveyor
    - allow running tests that need to change file permissions on Linux using chattr

    Instead of describing the error messages for legacy BDB wallets and descriptor SQLite wallets in a comment, check that the error message is thrown at runtime and document that it shouldn't throw.

3. Update `feature_reindex_readonly.py` to use the file permission utilities.
4. Improve `wallet_multiwallet.py` coverage, while updating to use the file permission utilities, which enables the test to work on Linux and provides logging and error handling.  Also add a log entry for the test.

jonatack committed:
Update tool_wallet tests for unexpected writes to wallet

PR 15687 added test coverage in test/functional/tool_wallet.py to reproduce
unexpected writes to the wallet file, as described in issue 15608:

- wallet tool info unexpectedly writes to the wallet file if the wallet file permissions are read/write
- wallet tool info raises if the wallet file permissions are read-only

Update these tests for two changes since PR 15687:

- the CI migration away from Appveyor
- the addition of descriptor wallets

Instead of describing the error messages for legacy BDB wallets and descriptor
SQLite wallets in a comment, check that the error message is thrown at runtime
and document that it shouldn't throw.

fanquake commented:
> Will rebase once that PR is merged.

Mark as draft for now then, if based on another PR?

jonatack committed:
Add file permission utility functions to test framework

jonatack committed:
Update wallet_multiwallet.py to use file permission utils

which enables the test to work on linux and provides the logging and error
handling contained in the utils.

Also add a log entry for the test.

jonatack committed:
Update feature_reindex_readonly.py to use file permission utils
